examples/segmentation/train_deeplabv3_ade20k.py

```diff
@@ -0,0 +1,359 @@
+
+    def combo_loss(output, target):
+        loss = {}
+        loss['cross_entropy'] = soft_cross_entropy(output, target, ignore_index=-1)
+        loss['dice'] = dice_loss_fn(output, target)
+        loss['total'] = args.cross_entropy_weight * loss['cross_entropy'] + args.dice_weight * loss['dice']
+        return loss
+
+    # Training and Validation metrics to log throughout training
+    train_metrics = MetricCollection([CrossEntropy(ignore_index=-1), MIoU(num_classes=150, ignore_index=-1)])
+    val_metrics = MetricCollection([CrossEntropy(ignore_index=-1), MIoU(num_classes=150, ignore_index=-1)])
+
+    # Create a ComposerClassifier using the model, loss function, and metrics
+    composer_model = ComposerClassifier(module=model,
+                                        train_metrics=train_metrics,
+                                        val_metrics=val_metrics,
+                                        loss_fn=combo_loss)
+
+    logging.info('Built Composer DeepLabv3+ model\n')
+
+    logging.info('Building optimizer and learning rate scheduler')
+    # Optimizer
+    optimizer = DecoupledSGDW(composer_model.parameters(),
+                              lr=args.learning_rate,
+                              momentum=args.momentum,
+                              weight_decay=args.weight_decay)
+
+    # Only use a LR schedule if no recipe is specified or if the hot recipe was specified
+    lr_scheduler = None
+    if args.recipe_name is None or args.recipe_name == 'hot':
+        lr_scheduler = CosineAnnealingScheduler()
+
+    logging.info('Built optimizer and learning rate scheduler')
+
+    logging.info('Building callbacks: SpeedMonitor, LRMonitor, and CheckpointSaver')
+    speed_monitor = SpeedMonitor(window_size=50)  # Measures throughput as samples/sec and tracks total training time
+    lr_monitor = LRMonitor()  # Logs the learning rate
+
+    # Callback for checkpointing
+    checkpoint_saver = CheckpointSaver(folder=args.save_checkpoint_dir, save_interval=args.checkpoint_interval)
+    logging.info('Built callbacks: SpeedMonitor, LRMonitor, and CheckpointSaver\n')
+
+    # Recipes for training DeepLabv3+ on ImageNet in order of increasing training time and accuracy
+    # To learn about individual methods, check out "Methods Overview" in our documentation: https://docs.mosaicml.com/
+    logging.info('Building algorithm recipes')
+    if args.recipe_name == 'mild':
+        algorithms = [
+            ChannelsLast(),
+            EMA(half_life='1000ba', update_interval='10ba'),
+        ]
+    elif args.recipe_name == 'medium':
+        algorithms = [
+            ChannelsLast(),
+            EMA(half_life='1000ba', update_interval='10ba'),
+            SAM(rho=0.3, interval=2),
+            MixUp(alpha=0.2),
+        ]
+    elif args.recipe_name == 'hot':
+        algorithms = [
+            ChannelsLast(),
+            EMA(half_life='2000ba', update_interval='10ba'),
```

Comment:
Hot recipe uses `update_interval='1ba'` for EMA.

Reply:
Ooops, fixed!